### PR TITLE
VTK and VTU output with level number, material id, and subdomain ids

### PR DESCRIPTION
**Pull request by guidokanschat**

### CODE REVIEW

File: source/grid/grid_out.cc [hunk 1]
```diff
@@ -2422,32 +2422,43 @@ namespace
...
       }
+  }
 
-    return patches;
+  std::vector<std::string> triangulation_patch_data_names ()
+  {
+    std::vector<std::string> v(4);
+    v[0] = "level";
+    v[1] = "material";
+    v[2] = "leaf_subdomain";
```
Comment (tjhei) [inline source/grid/grid_out.cc]: I would prefer the name "subdomain" over "leaf_subdomain".

**Comment by tjhei** [comment]
1. Can you add a test with a distributed Triangulation that also distributes level subdomains?
2. Are (level) subdomains set to invalid/artificial going to be written as 4 billion something or as -1/-2? I would prefer the latter but I think you are currently producing the former.

**Comment by luca-heltai** [comment]
This could be also the right moment to add manifold id into the output...

**Comment by guidokanschat** [comment]
@luca-heltai It IS the right moment.

**Comment by tjhei** [comment]
Latest version looks good to me. Can anybody else take a look and merge? (I am biased by having looked over his shoulder)

**Comment by tjhei** [comment]
but it doesn't compile:
```
/ssd/deal-git/source/grid/grid_out.cc:2443:13: error: ‘data’ was not declared in this scope
```

**Comment by guidokanschat** [comment]
Just updated and ran tests on the same computer. Should be fixed now

**Comment by luca-heltai** [comment]
This looks good to me.

**Pull request merged**